### PR TITLE
net/frr: add route-maps to OSPF

### LINKED ISSUE #1213: net/frr: Feature Request - OSPF filtering routes
I needs to redistribute in my OSPF network only few _kernel_ and _connected_ routes

Reading the FRR manual and after some manual tests the simple way to filter them look like via `distribute-list` and `access-list`

According to my tests we need to define the `access-list` in the **main** block
```
access-list NAME permit IPV4-NETWORK
access-list NAME deny IPV4-NETWORK
```
so something like: 
```
access-list foo1 permit 10.10.1.0/24
access-list foo1 permit 10.30.0.0/24
access-list foo1 deny any
access-list foo2 permit 10.20.2.0/24
access-list foo2 deny any
```
http://docs.frrouting.org/en/latest/filter.html

Then define the `distribute-list` inside the **router ospf** block
```
distribute-list NAME out (kernel|connected|static|rip|ospf)
```
In my conf
```
router ospf
 distribute-list foo1 out kernel
 distribute-list foo2 out connected
!
```

http://docs.frrouting.org/en/latest/ospfd.html?highlight=%22distribute-list%22#ospf-distribute-list

### PR DESCRIPTION
**Pull request by mimugmail**
Closes #1213 (finally)

### CODE REVIEW

File: net/frr/src/opnsense/mvc/app/views/OPNsense/Quagga/ospf.volt [hunk 1]
```diff
@@ -127,6 +128,32 @@ POSSIBILITY OF SUCH DAMAGE.
...
+            </thead>
+            <tbody>
+            </tbody>
+            <tfoot>
+                <tr>
+                    <td colspan="5"></td>
+                <td>
+                    <button data-action="add" type="button" class="btn btn-xs btn-default"><span class="fa fa-plus"></span></button>
+                    <button type="button" class="btn btn-xs reload_btn btn-primary"><span class="fa fa-refresh reloadAct_progress"></span> {{ lang._('Reload Service') }}</button>
+                    </td>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/views/OPNsense/Quagga/ospf.volt]: fixed, copied from prefix lists which was also wrong

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/ospf.xml [hunk 2]
```diff
@@ -26,6 +26,12 @@
         <help><![CDATA[Select other routing sources, which should be redistributed to the other nodes.]]></help>
         <hint>Type or select a route source.</hint>
     </field>
+    <field>
+      <id>ospf.redistributemap</id>
+      <label>Redistribution Map</label>
+      <type>dropdown</type>
+      <help>Route Map to set for Redistribution.</help>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/ospf.xml]: Done

File: net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml [hunk 3]
```diff
@@ -47,6 +47,19 @@
                         <static>Statically configured routes</static>
                 </OptionValues>
         </redistribute>
+        <redistributemap type="ModelRelationField">
+                <Model>
+                        <template>
+                                <source>OPNsense.quagga.ospf</source>
+                                        <items>routemaps.routemap</items>
+                                        <display>name</display>
+                                        <group>name</group>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml]: Done

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <type>select_multiple</type>
+    <style>tokenize</style>
+    <allownew>true</allownew>
+    <help>Select the Prefix List.</help>
+  </field>
+    <field>
+    <id>routemap.set</id>
+    <label>Set</label>
+    <type>text</type>
+    <help>Free text field for your set, please be careful! You can set e.g. "local-preference 300" or "community 1:1" (http://www.nongnu.org/quagga/docs/docs-multi/Route-Map-Set-Command.html#Route-Map-Set-Command)</help>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml]: There is more to look for, other places have them too. Will take a look in a next PR

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    <type>text</type>
+    <help>Route-map ID between 10 and 99. Be aware that the sorting will be done under the hood, so when you add an entry between it get's to the right position</help>
+  </field>
+  <field>
+    <id>routemap.match2</id>
+    <label>Prefix List</label>
+    <type>select_multiple</type>
+    <style>tokenize</style>
+    <allownew>true</allownew>
+    <help>Select the Prefix List.</help>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml]: Removed

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<form>
+  <field>
+    <id>routemap.enabled</id>
+    <label>Enabled</label>
+    <type>checkbox</type>
+    <help>Enable / Disable</help>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFRouteMaps.xml]: Removed

File: net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml [hunk 7]
```diff
@@ -223,5 +236,46 @@
...
+                        <enabled type="BooleanField">
+                                <default>1</default>
+                                <Required>Y</Required>
+                        </enabled>
+                        <name type="TextField">
+                                <default></default>
+                                <Required>Y</Required>
+                        </name>
+                        <action type="OptionField">
+                                <default></default>
```
[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml]: I'd leave it like it is, as Required is Y it is ok to have an empty default in dialog. A default could lead users to to not think about what they are doing.

[the remark this answers is not in the document]
Comment (fabianfrz) [reply net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml]: Then just remove the default

[the remark this answers is not in the document]
Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/models/OPNsense/Quagga/OSPF.xml]: Removed :)

**Comment by fichtner** [comment]
Nice work, merged ❤️

**Pull request merged**